classes/services/PublicationService.inc.php

```diff
@@ -120,7 +120,7 @@ public function validatePublication($hookName, $args) {
 		$props = $args[2];
 
 		// Ensure that the specified series exists
-		if (isset($props['seriesId'])) {
+		if (isset($props['seriesId']) && ($props['seriesId']) > 0) {
```

Comment:
Hmm, I think that we need to try to solve this at the source. If someone passes  a value of `0`, that should return a validation error because it's an invalid value.

Let's see if we can figure out why a `0` is being passed for the empty select field. The value that is submitted with the form is calculated here: https://github.com/pkp/ui-library/blob/master/src/components/Form/Form.vue#L178-L185. That suggests to me that the Vue component is storing the selected value as a `0`.

You can try to use the Vue.js dev tools for your browser to inspect the value in the form before it is submitted. It seems like it should be an empty string, which should get submitted as `?seriesId=&...`. So we'll need to figure out why it's getting converted to an integer in the browser, or whether that's coming in when the form is loaded or what.

Reply:
(And a related heads-up that we'll need to get rid of `0` IDs in favour of `nulls` so that we can properly specify FOREIGN KEY constraints in the database using the new DB schema toolset.)

Reply:
I tried the the following methods without success to manipulate the o in vue side without success.

1.  Defined a default value null and empty string  for the  Fieldbase class.
https://github.com/pkp/ui-library/blob/master/src/components/Form/fields/FieldBase.vue#L33

1.  Set  the default value for value using data () property here
https://github.com/pkp/ui-library/blob/master/src/components/Form/fields/FieldSelect.vue#L90
Here I could  set up the currentValue using a cached- computed method, but the value seems to be untouched and always defaltls to 0. We can use the currentValue instead of value,  which is possible to manipulate, but I was not sure of the general implications.

1. Tried the server side aspect of sending a null value from the CatalogEntryForm in omp by setting the value to null, when the seriesId is 0.  Then also the 0 was set for the value in the vue.
https://github.com/pkp/omp/blob/master/classes/components/forms/publication/CatalogEntryForm.inc.php#L64

1. manually  edited the vue js core library’s  toNumber() function to be sure, no 0 values are passed through, which  was also not working as expected.

1. Somehow if I use the ui-library with the examples,  there it works using the following example.
https://github.com/pkp/ui-library/blob/master/src/docs/components/Form/helpers/field-select-country.js

1. Updated to  vue js to 2.6.12,  but the 0 could not be manipulated.

Therefore I  have the feeling it is related to one of the bugs in vue, eventhough  the discussions suggest, that is being fixed  for  version 2.

https://github.com/vuejs/vue/pull/4118

https://github.com/vuejs/vue/issues/3541

1. The only solution which worked was intercepting the json object, in the ajax call in the Form.vue,but it does not answer the core problem of getting rid of the 0 in the browser,
https://github.com/pkp/ui-library/blob/master/src/components/Form/Form.vue#L207

@NateWr may be  there can be an aspect, I have missed  which you may see, may be we can go through together next year and look for the solution.

Reply:
Uff, this looks tricky. The value is definitely getting converted from `string` to `null` when it is initialized in the Vue.js component. I can track the type change down to https://github.com/pkp/pkp-lib/blob/master/js/classes/VueRegistry.js#L41-L44 by adding the following line of code after it:

```
console.log(data, $.extend(true, {}, baseData, data, {id: id}), args); // eslint-disable-line
```

Having said that, I'm not able to reproduce the problem locally. Even when the value in the Vue.js object is `null`, the request does not get sent with a `0`. The request payload is:

```
datePublished=2020-12-04&seriesId=&seriesPosition=safasdf&categoryIds=&urlPath=&coverImage%5Ben_US%5D=
```

Can you remind me what are the conditions for reproducing this? I wonder if it is browser-specific.